src/commands/project/delete/source.ts

```diff
@@ -153,6 +158,10 @@ export class Source extends SfCommand<DeleteSourceJson> {
     if (this.flags['track-source']) {
       this.tracking = await SourceTracking.create({ org: this.org, project: this.project });
     }
+
+    if (!validateTests(this.flags['test-level'], this.flags.tests)) {
+      throw messages.createError('error.NoTestsSpecified');
```

Comment:
@WillieRuemmele I mentioned this wasn't being executed before but I was wrong.
Running `--test-level RunSpecifiedTests --tests` results in an error from oclif's flag validation (generic error).

This code works if you pass `--test-level RunSpecifiedTests` without `--tests <testname>`, and gives a better error msg to users.